Rework main README (#1476)
tektoncd/dashboard: README.md

# Tekton Dashboard

[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://github.com/kubernetes/dashboard/blob/master/LICENSE)
[![Go Report Card](https://goreportcard.com/badge/tektoncd/dashboard)](https://goreportcard.com/report/tektoncd/dashboard)

> Tekton Dashboard is a general purpose, web-based UI for [Tekton Pipelines](https://github.com/tektoncd/pipeline) and [Tekton triggers](https://github.com/tektoncd/triggers) resource.

Reviewer: ```suggestion
Tekton Dashboard is a general purpose, web-based UI for [Tekton Pipelines](https://github.com/tektoncd/pipeline) and [Tekton triggers](https://github.com/tektoncd/triggers) resources.
```

> It allows users to manage and view Tekton resources creation, execution, and completion.

Reviewer: ```suggestion
It allows users to manage and view Tekton resource creation, execution, and completion.
```

Among other things, the Tekton Dashboard supports:

> - Filtering resources by labels

Reviewer: ```suggestion
- Filtering resources by label
```

> - Realtime view of PipelineRuns and TaskRuns logs

Reviewer: ```suggestion
- Realtime view of PipelineRun and TaskRun logs
```

> - View resources details and YAML

Reviewer: ```suggestion
- View resource details and YAML
```

- Show resources for the whole cluster or limit visibility to a particular namespace
- Import resources directly from a git repository

> - Setup webhooks for automatically triggering pipelines on git events (github and gitlab supported)

Reviewer: that's the webhooks-extension, not the core dashboard

Author: fixed

> - Adding functionnalities through extensions, like automatically triggering pipelines on git events (github and gitlab supported)

Reviewer: ```suggestion
- Adding functionality through extensions, like automatically triggering pipelines on git events (GitHub and GitLab supported)
```

should we link to the webhooks-extension here?

![Dashboard UI workloads page](docs/dashboard-ui.png)

## Start using Tekton Dashboard

- Tekton Dashboard [install](./docs/install.md)
- Tekton Dashboard [extensions](./docs/samples/extension/README.md)
- Try out the [katacoda tutorial](https://katacoda.com/ncskier/scenarios/tekton-dashboard)
- Take a look at our [roadmap](./roadmap.md)
- Feature requests and bug reports welcome, please [open an issue](https://github.com/tektoncd/dashboard/issues/new/choose)

> ## Read the docs

Reviewer: `## Which version should I use?`

This is linked from https://github.com/tektoncd/dashboard/blob/master/docs/README.md#L58
and points at `https://github.com/tektoncd/dashboard#which-version-should-i-use`

Whichever heading we choose we should make sure the link works.

Author: fixed

> It is **strongly recommended** to use the **v0.6.1.4** release for Tekton Pipelines v0.11.x and v0.12.x and Tekton Triggers v0.4

Reviewer: `**v0.6.1.4** release or newer`

Author: fixed

- This is a **critically important security release**

> - The following versions are deprecated and should be used for **development or isolated usage only**

Reviewer: ```suggestion
- Earlier versions are deprecated and should be used for **development or isolated usage only**
```

The table below includes newer releases and master, so saying 'the following versions' isn't correct.

| [v0.5.1](https://github.com/tektoncd/dashboard/releases/tag/v0.5.1) | [Docs @ v0.5.1](https://github.com/tektoncd/dashboard/tree/v0.5.1/docs) |
| [v0.5.0](https://github.com/tektoncd/dashboard/releases/tag/v0.5.0) | [Docs @ v0.5.0](https://github.com/tektoncd/dashboard/tree/v0.5.0/docs) |
| [v0.4.1](https://github.com/tektoncd/dashboard/releases/tag/v0.4.1) | [Docs @ v0.4.1](https://github.com/tektoncd/dashboard/tree/v0.4.1/docs) |
| [v0.4.0](https://github.com/tektoncd/dashboard/releases/tag/v0.4.0) | [Docs @ v0.4.0](https://github.com/tektoncd/dashboard/tree/v0.4.0/docs) |
| [v0.3.0](https://github.com/tektoncd/dashboard/releases/tag/v0.3.0) | [Docs @ v0.3.0](https://github.com/tektoncd/dashboard/tree/v0.3.0/docs) |
| [v0.2.1](https://github.com/tektoncd/dashboard/releases/tag/v0.2.1) | [Docs @ v0.2.1](https://github.com/tektoncd/dashboard/tree/v0.2.1/docs) |
| [v0.2.0](https://github.com/tektoncd/dashboard/releases/tag/v0.2.0) | [Docs @ v0.2.0](https://github.com/tektoncd/dashboard/tree/v0.2.0/docs) |
| [v0.1.1](https://github.com/tektoncd/dashboard/releases/tag/v0.1.1) | [Docs @ v0.1.1](https://github.com/tektoncd/dashboard/tree/v0.1.1/docs) |

> | [v0.1.0](https://github.com/tektoncd/dashboard/releases/tag/v0.1.0) | [Docs @ v0.1.0](https://github.com/tektoncd/dashboard/tree/v0.1.0/docs) |

Reviewer: 0.6.1 was the first release with docs content in the `docs` folder (not counting the extensions content in `docs/samples` which was never really promoted given it alpha/pre-alpha state). I'd be inclined to cut down this list a bit since we're advising users to pick up 0.6.1.4 or newer and don't really need to show every patch release for older versions. We could also use this to present our compatibility chart, maybe something like:

### Which version should I use?

It is **strongly recommended** to use the **v0.6.1.4** release or newer for Tekton Pipelines v0.11.x and v0.12.x and Tekton Triggers v0.4.
- This is a **critically important security release**
- Earlier versions are deprecated and should be used for **development or isolated usage only**

| Dashboard | Docs | Pipelines | Triggers |
|---|---|---|---|
| [HEAD](#) | [Docs @ HEAD](#) | v0.11.x -<br />v0.13.x | v0.5.x |
| [v0.7.0](#) | [Docs @ v0.7.0](#) | v0.11.x<br />v0.12.x | v0.5.x |
| [v0.6.1.5](#) | [Docs @ v0.6.1.5](#) | v0.11.x<br />v0.12.x | v0.4.x |
| [v0.6.1.4](#) | [Docs @ v0.6.1.4](#) | v0.11.x | v0.4.x |
| [v0.6.0](#) |  | v0.11.x | v0.3.x |
| [v0.5.3](#) |  | v0.10.x | v0.3.x |
| [v0.5.2](#) |  | v0.10.x | v0.2.x |
| [v0.5.0](#) |  | v0.10.x | v0.1 |
| [v0.4.1](#) |  | v0.8.0 | v0.1 |
| [v0.2.1](#) |  | v0.7.0 |  |
| [v0.1.1](#) |  | v0.5.2 |  |

All versions from v0.5.0 support a read-only mode.

Author: no v0.6.1.6 ?

Reviewer: not sure how I missed 0.3.0
support for pipelines 0.8 - 0.9 and triggers 0.1

yes we should include 0.6.1.6 and 0.7.0 once they're ready (still marked as pre-release)

Author: ok, i guess v0.4.1 is for pipelines v0.9.0 then ?

Reviewer: updated comment above, we skipped pipelines 0.9 support: https://github.com/tektoncd/dashboard/issues/1154#issuecomment-600115090

Author: Thanks, fixed.
